src/chttpd/src/chttpd_auth_request.erl

```diff
@@ -34,7 +34,11 @@ authorize_request_int(#httpd{path_parts=[]}=Req) ->
 authorize_request_int(#httpd{path_parts=[<<"favicon.ico">>|_]}=Req) ->
     Req;
 authorize_request_int(#httpd{path_parts=[<<"_all_dbs">>|_]}=Req) ->
-    Req;
+   case config:get("chttpd", "admin_only_all_dbs", "false") of
+       "false" -> Req;
+       "true" -> require_admin(Req);
+       Else  -> couch_log:error("Invalid setting for admin_only_all_dbs: ~p. Must be true or false.", [Else])
```

Comment:
this will cause a badmatch or something in the response itself, no? In general I'm not in favour of error messages that only the couchdb administrator can see and not the person/agent making the request. 

Either don't have the clause, so we get a case_clause (and a 500 server error) or flip it around like;

"true" -> require_admin(Req);
_ -> Req

Reply:
@rnewson sorry for making my intentions not clear enough. I’d like to avoid people leaving `_all_dbs` open when they think they’ve closed it due to a config typo like `admin_only_all_dbs = True`.

So in my first attempt I’m allowing non-admin access is the config value is explicitly `"true"`, and require admin access at all other times.

In my second attempt, I matched `"true"` and `"false"`, but that would lead to a bad match, if the config value had a typo, so I wanted at least to leave a nice error message.

Reply:
@rnewson agreed, good point. I switched it back to true/false and added `:get_boolean` so at least we get a unified bad value thing from the config system, which then points to the right place.